### ISSUE TITLE
Question: SapMachine JRE creates .crypto file during execution?

### ISSUE THREAD

**Issue by M3tatron85** [SAP/SapMachine#811]
TL;DR: does SAPMachine JRE create a .crypto file during execution and where can I find more information?

Hello,

we are using SAP MII and today I wanted to test the SapMachine 11 (LTS) JRE for Windows x64 by running the SAP MII Workbench.

As I did this, the process C:\Windows\System32\svchost.exe was trying to write a file the extension .crypto (635D02A9D9XXXXXXXXX884B82B3BCDAEA.crypto) in the working directory. As this directory is synced to the company server and the server does not permit the writing of .crypto files (and also other file extensions) to prevent ransom ware attacks, the execution of the application was blocked.

I disconnected the system from the network. It got an extensive scan by the IT department and was reinstalled for security reasons. Nothing out of the ordinary was found.

Did this file really result from the JRE? Can I find more information somewhere? Do we need to set an exception for the JRE to run?

Thank you very much for your help!

Nino

**Comment by RealCLanger** [comment]
Hi Nino,

I'm not aware of SapMachine (or the OpenJDK) writing *.crypto files - but certainly that does not mean it doesn't happen. Also, the application or some framework library could possibly do so.

To figure out who or what is creating these kind of files, I'd recommend to run a procmon trace to see which process (and above all, which thread/stack) is responsible for creating them. If we have a stack we can then hopefully go back to the source code to see why it happens.

Best regards
Christoph

**Comment by M3tatron85** [comment]
Hi Christoph,

thank you very much for the fast answer. We are currently looking into the issue and are trying to create a setup where the described behavior can be triggered with a defined procedure. After this step, we should be able to start a deeper analysis.

Best regards
Nino

**Comment by RealCLanger** [comment]
Hi Nino,

sounds good. I'll close this issue for now. Please feel free to reopen it or add a comment, in case you have an update on your investigations which would necessitate additional support from SapMachine side.

Cheers
Christoph
